src/Client/LocalConnection.cpp

```diff
@@ -35,6 +35,9 @@ LocalConnection::LocalConnection(ContextPtr context_, bool send_progress_, bool
 
 LocalConnection::~LocalConnection()
 {
+    /// Last query may not have been finished or cancelled due to exception on client side.
+    if (state && !state->is_finished && !state->is_cancelled)
+       sendCancel();
```

Comment:
`sendCancel` looks like a function that may throw an exception, e.g. on network disconnect, may it?

Reply:
It's local connection, no network connections, in  `sendCancel ` we just cancel executors. But let's add try/catch just in case, it will be better, thanks